### PR TITLE
unify the SDC integrator code paths

### PR DESCRIPTION
**Pull request by zingale**
This removes a lot of the redundancy in the various actual_integrator_sdc.H files

### CODE REVIEW

File: integration/integrator_setup_sdc.H [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    }
+
+    // Fill in the initial integration state.
+
+    burn_to_int(state, int_state);
+
+    // Set the tolerances.
+
+    // we use 1-based indexing inside of the integrator, so we need to
+    // shift the indices SRHO, SFS, etc by 1
```
Comment (yut23) [inline integration/integrator_setup_sdc.H:59]: I don't think this comment applies anymore

Comment (zingale) [reply integration/integrator_setup_sdc.H:59]: indeed, you are right.  I think that's from back when the tolerances were arrays

Comment (yut23) [reply integration/integrator_setup_sdc.H:59]: The manual indexing was removed in https://github.com/AMReX-Astro/Microphysics/pull/1039

Comment (zingale) [reply integration/integrator_setup_sdc.H]: fixed

**Review by yut23 — APPROVED** [review]

**Pull request merged**